### PR TITLE
Add cronjob for running cleanup command

### PR DESCRIPTION
**Pull request by HappyTetrahedron**
## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: .cruft.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
       "name": "APPUiO Control API",
       "slug": "control-api",
       "parameter_key": "control_api",
-      "test_cases": "defaults insecure",
+      "test_cases": "defaults insecure withcronjob",
```
Comment (simu) [inline .cruft.json:10]: Some boilerplate updates (GitHub actions) are missing, did you use `commodore component update -t withcronjob` to generate the new test case?

File: class/defaults.yml [hunk 2]
```diff
@@ -130,6 +130,18 @@ parameters:
...
+      enabled: false
+      schedule: '45 0 * * *'
+      extraArgs: []
+      extraEnv: {}
+      nodeSelector:
+        node-role.kubernetes.io/infra: ''
+      jobHistoryLimit: {
+        failed: 2,
+        successful: 2,
+      }
```
Comment (simu) [inline class/defaults.yml:143]:
```suggestion
      jobHistoryLimit: {
        failed: 2,
        successful: 2,
      }
```
```suggestion
      jobHistoryLimit:
        failed: 2
        successful: 2
```

File: component/cleanup-cronjob.libsonnet [hunk 3]
```diff
@@ -0,0 +1,96 @@
+/*
+* Deploys the control-api API server
```
Comment (simu) [inline component/cleanup-cronjob.libsonnet:2]: Should probably be something like

```suggestion
* Deploys the control-api billing entity cleanup cronjob
```

File: component/cleanup-cronjob.libsonnet [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+local countriesConfigMap =
+  kube.ConfigMap('billing-entity-odoo8-country-list') {
+    metadata+: {
+      namespace: params.namespace,
+    },
+    data: {
+      'billing_entity_odoo8_country_list.yaml': std.manifestYamlDoc(countryList),
+    },
+  };
```
Comment (simu) [inline component/cleanup-cronjob.libsonnet:46]: As far as I can see, this is unnecessary copy-pasted logic, as we already create the exact same object in `component/api-server.jsonnet` and you don't even emit the `countriesConfigMap` object in this file.

```suggestion
```

File: component/cleanup-cronjob.libsonnet [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                        volumes+: [
+                          {
+                            name: 'countries-config',
+                            configMap: {
+                              name: countriesConfigMap.metadata.name,
+                            },
+                          },
+                        ],
+                      }
+                    else {}),
```
Comment (simu) [inline component/cleanup-cronjob.libsonnet:89]: You can do the following with `kube.libsonnet`:

```suggestion
                 + (if hasCountriesConfig then
                      {
                        volumes_: {
                          'countries-config': {
                            configMap: {
                              name: countriesConfigMap.metadata.name,
                            },
                          },
                      }
                    else {}),
```

Alternatively, you can write this as 

```suggestion
                 [if hasCountriesConfig then 'volumes_']: {
                    'countries-config': {
                      configMap: {
                        name: countriesConfigMap.metadata.name,
                      },
                    },
                 },
```

**Review by simu — CHANGES_REQUESTED** [review]

File: .cruft.json (hunk 1 above)
Comment (HappyTetrahedron) [reply .cruft.json:10]: I didn't. I didn't know I could do that, haha.

File: component/cleanup-cronjob.libsonnet [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        template+: {
+          spec+: {
+            nodeSelector: params.cleanupJob.nodeSelector,
+            restartPolicy: 'Never',
+            containers_+: {
+              cleanup: kube.Container('cleanup') {
+                image: '%(registry)s/%(image)s:%(tag)s' % params.images['control-api'],
+                args: common.MergeArgs(extraDeploymentArgs, params.cleanupJob.extraArgs),
+                env+: com.envList(params.cleanupJob.extraEnv),
+                volumeMounts+:
```
Comment (simu) [inline component/cleanup-cronjob.libsonnet:45]: I didn't notice yesterday but IIRC there's a `volumeMounts_` which helps you a bit (similar to `volumes_`). I don't remember off the top of my head how it works, so I can't make a suggestion. If you want you look it up you can adjust the code, but I don't mind either way.

**Review by simu — APPROVED** [review]

**Pull request merged**